### PR TITLE
Wrestle&Sprint Tackle Nerf: the Salt PR

### PR DESCRIPTION
**Pull request by CormanQuewye**
## About The Pull Request
Are you sick of high CON characters toggling run and rushing you like a dark fantasy ballistic missile to win every fight at no risk? So am I!
- Sprinting into someone is no longer a simple check on if you have higher CON and STR. It'll now compare the sprinter's CON and athletics to their adversary's END and athletics. At equal value, they have a 30% chance to knock their adversary down;
- If you have combat mode on, you will no longer drop held items on knockdown;
- Sprinting over corpses/lying down mobs will result in you tripping.

## Why It's Good For The Game
There should be not tactic in combat that is totally lacking in risk. Currently, sprinting fits the bill. If you're playing a high con character and you're aware your adversary's race and class has them at lower con and str, you can simply sprint into them. This'll cause them to fall prone, take damage, and drop their items, making them an incredibly easy target. Very low risk and very high reward.

Wrestling is in a similar place where being able to grab someone's chest and tackle them will result in a variety of high value gains (placing the enemy prone and disarming them) so even if you want to repeatedly stab the guy that thought it'd be a good idea to get into a grappling match with someone holding a dagger, that won't be happening.

To resolve the issue of sprint tackling, it now performs a check of constitution and athletics versus endurance and athletics, weighted towards the defender, with a base success rate of 30%, minimum of 10%, and maximum of 90%. If you're haphazardly barrelling towards the enemy in a heated fight, you shouldn't expect things to go well for you. The reason for why I chose endurance as the defender's checked stat is primarily because too many opposed checks are mirror checks of either strength or constitution, this is an attempt to divert from that logic.

Furthermore, I added sprint tripping over lying down mobs for two reasons: to discourage sprint-charging in messy battlefields that are littered with bodies and to reward quick-drawing defenders that are able to quickly lie down in response to a charging opponent.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by CormanQuewye** [comment]
I somehow missed that another PR was up that affects sprint charging, but after looking through its code I have issues with it which can be found in the comment I left on it! https://github.com/Darkrp-community/OpenKeep/pull/579

**Comment by YuiY1997** [comment]
I like the sprint changes, I'm not a fan of the grappling changes.

**Comment by CormanQuewye** [comment]
> I like the sprint changes, I'm not a fan of the grappling changes.

There are no 'grapple changes' per say, I just made it so everyone doesn't have butter fingers and don't have stuff in their hands flying out on knockdown, if they have combat mode on. Grapple is affected indirectly as pushing someone to the ground is a knockdown.

But arguing on action economy: if you're playing a class with high wrestling, all you need to do is walk to your opponent, grab chest, shove. They are knocked down, locking them from acting for a few seconds and also lose their weapon. With two actions you've accomplished:

- Knocking your opponent down;
- Disarming them;
- Stunning them.

Your options are now several:
- Pin;
- Kick/Stomp;
- Twist a different bodypart;
- Throw;
- Stab with a weapon.
- Etc.

Your opponent's options are:
- Resist.

If they do not resist they risk getting pinned and/or tied up. If they do resist, it's an opposed check and if your opponent is starting off with a grapple tackle then it's likely they have high wrestling and know they can win. Even if you do succeed at resisting, resisting alone takes more stamina than applying the grapple and there's no delay, meaning the grapple can just be re-applied leading to a stamina deathspiral.

This is very comparable to how maces used to be, where with high maces a single click would achieve multiple things that would put your opponent in a position where they had practically no options and all of them were bad while the person with the mace had a variety of high reward options for little-to-no risk.

**Comment by Cre77** [comment]
> I somehow missed that another PR was up that affects sprint charging, but after looking through its code I have issues with it which can be found in the comment I left on it! #579

i stand by my preference of ratwood's handling of it, but  i very much enjoy the tripping changes here and ESPECIALLY knockdown, since a lot of fights can be cheesed by tackling/grappling/hitting someone with a big enough stick and just stealing their weapon.

could probably get optimism to decide on this, i trust his sense of balance

**Comment by CormanQuewye** [comment]
> You seemed to have it set to adding the defenders unarmed skill towards their tackle defense equation, massively skewing the equation in favor of the defender. Even without that, I feel a 30% base chance is really bad considering you can no longer knock their weapons out of their hand and have more obstacles to worry about. It should really just be a even 50%.

Thanks for mentioning! I thought I deleted that but I must've forgotten. An even 50% would mean that Barbarian would easily hit that 90% cap against non combatants when it's meant to be an unhinged and undisciplined move. And this isn't even taking into account bonuses from drugs, racials, abilities, etc. Even at 30%, there are ways of skewing it in your favour.

Yes, they no longer lose their weapon so it's not instant death. They're still prone, which applies a big defensive penalty. If you want a safer way of achieving the same, that's why grab chest > tackle is for.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -184,15 +184,18 @@
...
-					L.Knockdown(1)
-				else
-					Knockdown(1)
-			if(STACON < L.STACON)
-				Knockdown(30)
-			if(STACON == L.STACON)
-				L.Knockdown(1)
+			var/attacker_skill = STACON * 10
+			var/defense_skill = L.STAEND * 10
+			var/tacklechance = 30 - (defense_skill - attacker_skill)
```
Comment (ThePainkiller) [inline code/modules/mob/living/living.dm:189]: Wouldn't this make 30% chance the ceiling? The math could do some tuning, much as a fan I am of pitting CON vs END and athletics difference as well

**Comment by CormanQuewye** [comment]
> > > You seemed to have it set to adding the defenders unarmed skill towards their tackle defense equation, massively skewing the equation in favor of the defender. Even without that, I feel a 30% base chance is really bad considering you can no longer knock their weapons out of their hand and have more obstacles to worry about. It should really just be a even 50%.
> > 
> > 
> > Thanks for mentioning! I thought I deleted that but I must've forgotten. An even 50% would mean that Barbarian would easily hit that 90% cap against non combatants when it's meant to be an unhinged and undisciplined move. And this isn't even taking into account bonuses from drugs, racials, abilities, etc. Even at 30%, there are ways of skewing it in your favour.
> > Yes, they no longer lose their weapon so it's not instant death (and this goes both ways, you won't lose your weapon if your charge fails and you're knocked down). They're still prone, which applies a big defensive penalty. If you want a safer way of achieving the same, that's what grab chest > tackle is for.
> > At an even 50% it might as well just be the current system with a 1-in-10 chance for a crit fail because it'd still just be roles with high con purely using it against stuff like rogues because they know they have it beat. I feel like a good way of thinking about this is comparing sprint tackle to other stuff that achieves the same (knockdown).
> > Grapple requires you to land the grab and manage to push them over after; kick and push need you to shove them into an obstacle; hits typically require a crit. Sprint tackle? You don't have to worry about your opponent's defenses, you don't have to worry about rolls, you don't even need to click anything just run into your opponent. Since it's so easy to pull off, it should comparatively be riskier than wrestling or shove/kicking.
> > For a bit of numerical reasoning: barbarian has 3 athletics and +2 CON so your average humen barbarian has a total of 15, your average humen which is what we should be comparing balance to has 10 in every stat. So, barbarian has +50. At 50, that means they'd just be hitting the cap. At 30, they'd still be doing fine at 80% chance of success. But, heck, let's compare to an adventurer! Rogue has Athletics 4 and no END mod. So, barbarian would have... a 40% chance of success. I'd say that's not bad, considering the amount of input and potential reward. And reminder that this goes both ways, if someone is charging you they're pulling something risky off. I don't think we need 'whoever charges the other first' fights.
> > tl;dr sprint tackle ignores a bunch of the game's mechanics so it's not supposed to really be all that good
> 
> tons of jobs and even non combat classes have at least average level athletics, and also +END mods are very common as well, so I feel as if you are portraying the average human wrong. Even more so that you're comparing them to a barbarian who realistically SHOULD be knocking people over because they're one of the strongest and most athletic classes out there.

They still can knock people over utilizing grapple tackles or shoves. The issue is sprint tackling is so low effort, ignores so many mechanics, and in other words: is so MUCH better than the other knockdown options that I feel people don't even consider them when discussing sprint charge changes.

**Comment by ThePainkiller** [comment]
Guys please trim your replies to the latest one, and please address the fact it's only a 30% ceiling unless my math is off

**Comment by CormanQuewye** [comment]
> Guys please trim your replies to the latest one, and please address the fact it's only a 30% ceiling unless my math is off

I responded to your review. That's not how math works, Nick. v.v

**Comment by CormanQuewye** [comment]
I'll likely be closing this PR and opening a new one with purely the changes to knockdown and tripping over people, since that got good reception overall and generally the community seems more receptive to the modified Ratwood port of sprint charging! I'm glad you could all join me in finding out Nick cannot do basic math. :^)

**Comment by Optimism333** [comment]
> I'll likely be closing this PR and opening a new one with purely the changes to knockdown and tripping over people, since that got good reception overall and generally the community seems more receptive to the modified Ratwood port of sprint charging! I'm glad you could all join me in finding out Nick cannot do basic math. :^)

Don't be rude.

**Comment by CormanQuewye** [comment]
> > I'll likely be closing this PR and opening a new one with purely the changes to knockdown and tripping over people, since that got good reception overall and generally the community seems more receptive to the modified Ratwood port of sprint charging! I'm glad you could all join me in finding out Nick cannot do basic math. :^)
> 
> Don't be rude.

'twas joke. I like Nick. Madlad made the Malum PR. v.v
